src/libraries/System.Private.CoreLib/src/System/Number.NumberBuffer.cs

```diff
@@ -16,6 +16,7 @@ internal static partial class Number
         internal const int Int32NumberBufferLength = 10 + 1;    // 10 for the longest input: 2,147,483,647
         internal const int Int64NumberBufferLength = 19 + 1;    // 19 for the longest input: 9,223,372,036,854,775,807
         internal const int SingleNumberBufferLength = 112 + 1 + 1;  // 112 for the longest input + 1 for rounding: 1.40129846E-45
+        internal const int HalfNumberBufferLength = 21;
```

Comment:
This should be `21 + 1 + 1` with a similar comment around +1 for rounding (and then +1 for the null terminator)